tidy3d/plugins/mode/mode_solver.py

```diff
@@ -579,6 +579,104 @@ def to_mode_solver_monitor(self, name: str) -> ModeSolverMonitor:
             name=name,
         )
 
+    def add_source(
+        self,
+        source_time: SourceTime,
+        direction: Direction,
+        mode_index: pydantic.NonNegativeInt = 0,
+        ) -> Simulation:
+
+        """Creates :class:`Simulation` from a :class:`ModeSolver` instance plus additional
+        specifications.
+
+        Parameters
+        ----------
+        source_time: :class:`.SourceTime`
+            Specification of the source time-dependence.
+        direction : Direction
+            Whether source will inject in ``"+"`` or ``"-"`` direction relative to plane normal.
+        mode_index : int = 0
+            Index into the list of modes returned by mode solver to use in source.
+
+        Returns
+        -------
+        :class:`.Simulation`
+            Copy of the simulation with a mode source with specifications taken
+            from the ModeSolver instance and the method inputs.
+        """
+
+        mode_source = self.to_source(mode_index=mode_index, direction=direction, source_time=source_time)
+        new_sources = self.simulation.sources
+        if not new_sources:
+            new_sources = [mode_source]
+        else:
+            new_sources = new_sources + tuple([mode_source])
+        newSim = self.simulation.copy(update=dict(sources=new_sources))
+
+        return newSim
+
+    def add_monitor(
+        self,
+        mode_monitor_freqs: List[float],
```

Comment:
Let's keep these arguments to be called the exact same as the wrapped methods, eg `freqs` and `name`. I think it might be confusing to change them.

For example, a user could have these two lines in their code
```py
monitor = mode_solver.to_monitor(freqs=freqs, name=name)
sim = mode_solver.simulation.updated_copy(monitors=[monitor])
```

and wouldn't need to look up the new argument names and instead just write
```
sim = mode_solver.sim_with_monitor(freqs=freqs, name=name)
```